core/ai_worker.go

```diff
@@ -0,0 +1,991 @@
+		return nil, err
+	}
+
+	res, err = orch.node.saveRemoteAIWorkerResults(ctx, res, requestID)
+	if err != nil {
+		clog.Errorf(ctx, "Error saving remote ai result err=%q", err)
+		if monitor.Enabled {
+			monitor.AIResultSaveError(ctx, "image-to-video", *req.ModelId, string(monitor.SegmentUploadErrorUnknown))
+		}
+		return nil, err
+	}
+
+	return res.Results, nil
+}
+
+func (orch *orchestrator) Upscale(ctx context.Context, requestID string, req worker.UpscaleMultipartRequestBody) (interface{}, error) {
+	//local AIWorker processes job if combined orchestrator/ai worker
+	if orch.node.AIWorker != nil {
+		workerResp, err := orch.node.Upscale(ctx, req)
+		if err == nil {
+			return orch.node.saveLocalAIWorkerResults(ctx, *workerResp, requestID, "image/png")
+		} else {
+			clog.Errorf(ctx, "Error processing with local ai worker err=%q", err)
+			if monitor.Enabled {
+				monitor.AIResultSaveError(ctx, "upscale", *req.ModelId, string(monitor.SegmentUploadErrorUnknown))
+			}
+			return nil, err
+		}
+	}
+
+	//remote ai worker proceses job
+	imgBytes, err := req.Image.Bytes()
+	if err != nil {
+		return nil, err
+	}
+
+	input_url, err := orch.SaveAIRequestInput(ctx, requestID, imgBytes)
+	req.Image.InitFromBytes(nil, input_url)
+
+	res, err := orch.node.AIWorkerManager.Process(ctx, requestID, "upscale", *req.ModelId, input_url, req)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err = orch.node.saveRemoteAIWorkerResults(ctx, res, requestID)
+	if err != nil {
+		clog.Errorf(ctx, "Error saving remote ai result err=%q", err)
+		if monitor.Enabled {
+			monitor.AIResultSaveError(ctx, "upscale", *req.ModelId, string(monitor.SegmentUploadErrorUnknown))
+		}
+		return nil, err
+	}
+
+	return res.Results, nil
+}
+
+func (orch *orchestrator) AudioToText(ctx context.Context, requestID string, req worker.AudioToTextMultipartRequestBody) (interface{}, error) {
+	//local AIWorker processes job if combined orchestrator/ai worker
+	if orch.node.AIWorker != nil {
+		//no file response to save, response is text sent back to gateway
+		return orch.node.AudioToText(ctx, req)
+	}
+
+	//remote ai worker proceses job
+	audioBytes, err := req.Audio.Bytes()
+	if err != nil {
+		return nil, err
+	}
+
+	input_url, err := orch.SaveAIRequestInput(ctx, requestID, audioBytes)
+	req.Audio.InitFromBytes(nil, input_url)
+
+	res, err := orch.node.AIWorkerManager.Process(ctx, requestID, "audio-to-text", *req.ModelId, input_url, req)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err = orch.node.saveRemoteAIWorkerResults(ctx, res, requestID)
+	if err != nil {
+		clog.Errorf(ctx, "Error saving remote ai result err=%q", err)
+		if monitor.Enabled {
+			monitor.AIResultSaveError(ctx, "audio-to-text", *req.ModelId, string(monitor.SegmentUploadErrorUnknown))
+		}
+		return nil, err
+	}
+
+	return res.Results, nil
+}
+
+func (orch *orchestrator) SegmentAnything2(ctx context.Context, requestID string, req worker.SegmentAnything2MultipartRequestBody) (interface{}, error) {
```

Comment:
That is correct and is from the initial architecture of the AI pipelines.  My understanding is that the main driver of this is the different return types and request data structs (including request input file names).  This could pivot to something like `handleAIRequest` but that just trades separate functions for a longer function.

If ai-worker was updated to have a higher level input request struct and results return type I think the go-livepeer side of things could be generalized more.   

Nico's version is shorter because it passes the base64 encoded result up to the gateway which is then decoded by the gateway and saved to storage to provide a url to download to the client.

Reply:
ok, fine. We can leave it for now and revisit if we start having many more AI Worker endpoints (as discussed in Discord).